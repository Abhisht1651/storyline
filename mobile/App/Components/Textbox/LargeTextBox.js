// @flow
import React, { useState, useEffect } from 'react'
import { View, TextInput, TouchableOpacity, KeyboardAvoidingView } from 'react-native'
import styles from '../Styles/TextboxStyles/LargeTextBoxStyle'
import Sticker from '../../Images/Icons/icon-sticker.svg'
import Colors from '../../Themes/Colors'

type Props = {
  textColor: string,
  placeholder: string,
  placeholderColor: string,
  captureText: (args: string) => void
}
const LargeTextBox = (props: Props) => {
  const [value, setValue] = useState('')
  const { textColor, placeholder, placeholderColor, captureText } = props

  const textColorStyle = {
    color: textColor
  }

  useEffect(() => {
    captureText(value)
  }, [value])

  const changeTextHandler = (text: string) => {
    setValue(text)
  }

  return (
    <KeyboardAvoidingView behavior="padding">
      <View style={styles.container}>
        <View style={styles.innerContainer}>
          <TextInput
            placeholder={placeholder}
            multiline
            placeholderTextColor={placeholderColor}
            spellCheck={false}
            autoCorrect={false}
            value={value}
            selectionColor={Colors.cursor}
            style={[styles.textInput, textColorStyle]}
            onChangeText={changeTextHandler}
          />
        </View>
        <View style={styles.stickerContainer}>
          <TouchableOpacity style={styles.sticker} activeOpacity={1}>
            <Sticker />
          </TouchableOpacity>
        </View>
      </View>
    </KeyboardAvoidingView>
  )
}

LargeTextBox.defaultProps = {
  textColor: '#000080',
  placeholder: 'You must write something to be able to post!',
  placeholderColor: '#8F8FAF'
}

export default LargeTextBox
