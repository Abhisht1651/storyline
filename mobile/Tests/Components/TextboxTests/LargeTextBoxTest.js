import { shallow } from 'enzyme'
import React from 'react'
import { TextInput } from 'react-native'
import LargeTextBox from '../../../App/Components/Textbox/LargeTextBox'

describe('LargeTextBox', () => {
  it('should match to snapshot', () => {
    const component = shallow(<LargeTextBox />)
    expect(component).toMatchSnapshot()
  })
  it('should store the text when text is entered', () => {
    const component = shallow(<LargeTextBox />)
    component.find(TextInput).prop('onChangeText')('Green fox jumps over the wall')
    expect(component.find(TextInput).prop('value')).toEqual('Green fox jumps over the wall')
  })
})
