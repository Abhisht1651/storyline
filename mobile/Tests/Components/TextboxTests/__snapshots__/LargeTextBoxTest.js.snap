// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeTextBox should match to snapshot 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(143, 143, 175,0.24)",
        "borderRadius": 10,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginLeft": 40,
          "marginTop": 49.285714285714285,
          "maxWidth": "73.76%",
        }
      }
    >
      <Component
        allowFontScaling={true}
        autoCorrect={false}
        multiline={true}
        onChangeText={[Function]}
        placeholder="You must write something to be able to post!"
        placeholderTextColor="#8F8FAF"
        rejectResponderTermination={true}
        selectionColor="#7B46E4"
        spellCheck={false}
        style={
          Array [
            Object {
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 24,
              "height": "100%",
              "letterSpacing": -0.11,
              "paddingTop": 0,
              "textAlignVertical": "top",
            },
            Object {
              "color": "#000080",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "height": 60,
          "paddingBottom": 15,
          "paddingRight": 15,
        }
      }
    >
      <ForwardRef
        activeOpacity={1}
        style={
          Object {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#7B46E4",
            "borderRadius": 50,
            "height": 45,
            "justifyContent": "center",
          }
        }
      >
        <SvgMock />
      </ForwardRef>
    </View>
  </View>
</KeyboardAvoidingView>
`;
